tests/src/test/java/org/wildfly/plugins/bootablejar/maven/goals/AbstractDevWatchTestCase.java

```diff
@@ -94,7 +94,11 @@ public void run() {
                     } else {
                         cmd.add("mvn");
                     }
-                    String[] mvnCmd = {"-f", pomFile.toAbsolutePath().toString(), "wildfly-jar:dev-watch", "-e", prop};
+                    String localRepository = System.getProperty("maven.repo.local");
+                    String[] mvnCmd = {"-f", pomFile.toAbsolutePath().toString(), "wildfly-jar:dev-watch", "-e", prop,
+                            // need to use the same repository as it is used for build
+                            localRepository != null && !localRepository.isEmpty() ? "-Dmaven.repo.local=" + localRepository : ""
```

Comment:
In case the localRepository is null, then we add an empty "" in the array. You could remove this option from the array. Keep the array as it is, then do:
cmd.addAll(Arrays.asList(mvnCmd));
if(localRepository != null && !localRepository.isEmpty()) {
  cmd.add("-Dmaven.repo.local=" + localRepository);
}

Reply:
okay, updated